File: src/frontend/src/models/TestDto.ts
import EvaluationDto from './EvaluationDto'

// DTO for Test evaluations
export default class TestDto extends EvaluationDto {
  constructor(
    title: string,
    date: string,
    weight: number,
    curricularUnitId: number,
    revisionDeadline: string,
    id?: number
  ) {
    super(title, date, weight, curricularUnitId, 'TEST', revisionDeadline, id)
  }

  // Factory method to create from backend response
  static fromBackend(data: any): TestDto {
    return new TestDto(
      data.title,
      data.date,
      data.weight,
      data.curricularUnitId,
      data.revisionDeadline,
      data.id
    )
  }
}

File: src/backend/src/main/java/pt/ulisboa/tecnico/rnl/dei/dms/evaluation/dto/TestDto.java
package pt.ulisboa.tecnico.rnl.dei.dms.evaluation.dto;

import pt.ulisboa.tecnico.rnl.dei.dms.evaluation.domain.Test;

import java.time.LocalDateTime;

// Data Transfer Object for Test evaluations
public record TestDto(
    Long id,
    String title,
    LocalDateTime date,
    Double weight,
    Long curricularUnitId,
    String curricularUnitName,
    boolean isCompleted,
    boolean isUpcoming
) {
    public TestDto(Test test) {
        this(
            test.getId(),
            test.getTitle(),
            test.getDate(),
            test.getWeight(),
            test.getCurricularUnit().getId(),
            test.getCurricularUnit().getName(),
            test.isCompleted(),
            test.isUpcoming()
        );
    }
}
